### DISCUSSION TITLE
disabling compression in minio server

### DISCUSSION THREAD

**Discussion by OmerPe** [minio/minio#18440] [Q&A] [answered]
hey, i was trying out minio for saving large amounts of data with emphasis on performance, 
to do it i created said data and saved it once to the bucket, using minio service in my cluster.
and once straight to the disk.

i noticed in the bucket that the size of it is much smaller than the size of the files on the disk, and it changes from run to run (the data is randomized but have the same size)
so i assume minio compresses the data while saving it to the disk. 

i tried disabling the compression while testing i dont see any difference. also looking at the GUI the default compression file types are still there. 
the way I've tried doing so is by using the env variables before starting the server: 
(yaml file config)
`      containers:
        - name: minio-dev
          image: quay.io/minio/minio:latest
          env:
            - name: MINIO_COMPRESSION_ENABLE
              value: "false"
            - name: MINIO_COMPRESSION_EXTENSIONS
              value: ""
            - name: MINIO_COMPRESSION_MIME_TYPES
              value: ""
          command:
            - /bin/bash
            - -c
          args:
            - minio server /data --console-address :9080`
           
also tried using only the "MINIO_COMPRESSION_ENABLE" flag without the lists, tried adding the empty lists to see the diff in the GUI.. (also i used "off" and other ways of disabling the flag.. still didn't work) 

from the documentation is looks like this will be the optimal way to make sure i don't run compression, since the docs says that env variables override other configurations, but for some reason it doesn't seem to affect anything.
what am i missing ?

**Comment by klauspost** [comment]
There is no compression unless you enabled it.

**Reply by OmerPe** [reply]
in that case, what might cause the difference I see between the size of the bucket and the size on the disk ? 

edit:
for clarification, on the bucket i see a much smaller volume, sometimes 5 times smaller than the same files on the disk
and it takes 6+ times longer to save to the bucket than to save to the disk..

**Reply by klauspost** [reply, marked as answer]
You provide no real information on your setup. Also bucket sizes are not instantly updated.

**Reply by OmerPe** [reply]
The setup is basically the "quick start guide" setup for kubernetes, but instead of creating a single pod im creating a deployment that deploys a single pod and a NodePort service, and using a python script (also the similar to the quick start guide) that basically generates a lot of numpy.ndarrays and saves them to the bucket and to the disk (for comparison) as ".npy" files.. also saving some randomly generated csv/json files.

saving to the bucket i used BytesIO stream, wrote the numpy array to there and passed it to minio_client.put_object
`stream = io.BytesIO()
    np.save(stream, nd_array)
    stream.seek(0)
    minio_client.put_object(bucket_name, file_name, stream, len(stream.getbuffer()))`

Anyways, I ran it again to check, after it was done it showed ~5GiB, and ~10 minutes later it updated to 14.2GiB, on linux it shows that the files are 15.2GB but its close enough to where it might be a difference of how the system presents the size..

Tbh I would've loved a visual way to see if compression (and i guess other similar features) are on/off, the only indication in the GUI settings is the files types to be compressed.. which made me think its on.

thank you very much!
